### ISSUE TITLE
Problem with localdata in case of multiple search fields

### ISSUE THREAD

**Issue by hardiksondagar** [ghiden/angucomplete-alt#330]
## Problem Case ##
The local-data `userslist` have one record with value `{ first_name:'Elon', last_name: 'Musk'}`. When I search with `Elon` or `Musk` the autocomplete showing correct results. But when I search with `Elon Musk` it's not working because in code search query is compared to individual search fields. 

Is there any option to enable comparison with combined search value ( means search query "Elon Musk" will compare with combined search value first_name+last_name from the local-data ).
```
<angucomplete-alt 
 id="ex1"
 placeholder="Type the name of a person"
 pause="400"
 selected-object="selectedUser"
 local-data="userslist"
 search-fields="first_name,last_name"
 title-field="first_name,last_name"
minlength="2"
input-changed="inputChangeHandler"
input-class="form-control form-control-small"/>
```

In case there is no such option, here is my small fix. 
Updated `getLocalResults` function.
```
 function getLocalResults(str) {
          // combinedValue will store concatenated search values from local-data
          var i, match, s, value, combinedValue, 
          searchFields = scope.searchFields.split(','),
          matches = [];
          if (typeof scope.parseInput() !== 'undefined') {
            str = scope.parseInput()(str);
          }
          for (i = 0; i < scope.localData.length; i++) {
            match = false;

            combinedValue = '';
            for (s = 0; s < searchFields.length; s++) {
              value = extractValue(scope.localData[i], searchFields[s]) || '';
              match = match || (value.toString().toLowerCase().indexOf(str.toString().toLowerCase()) >= 0);
              combinedValue = combinedValue + value;
            }
            
            match = match || (combinedValue.toString().toLowerCase().replace(/ /g,'').indexOf(str.toString().toLowerCase().replace(/ /g,'')) >= 0);
            
            if (match) {
              matches[matches.length] = scope.localData[i];
            }
          }

          scope.searching = false;
          processResults(matches, str);
        }
```

**Comment by ghiden** [comment]
Try this branch:
https://github.com/ghiden/angucomplete-alt/tree/custom-local-search

I've added a custom local search handler.
With this, you don't need search-fields anymore but you need to provide a function that does your custom search. All it needs to do is given a user input string, return an array containing all the matches. If there is no match, you still need to return an empty array.

```html
<angucomplete-alt 
  id="ex1"
  placeholder="Type the name of a person"
  pause="400"
  selected-object="selectedUser"
  local-data="userslist"
  local-search="localSearch"
  title-field="first_name,last_name"
  minlength="2"
  input-changed="inputChangeHandler"
  input-class="form-control form-control-small"/>
```

Probably something like this, in your case:
```javascript
$scope.localSearch = function(str) {
  var matches = [], firstName, lastName, fullName, user;
  $scope.userlist.forEach(function(user) {
    if ((user.first_name.toLowerCase().indexOf(str.toString().toLowerCase()) >= 0) ||
        (user.last_name.toLowerCase().indexOf(str.toString().toLowerCase()) >= 0) ||
        ((user.first_name + ' ' + user.last_name).toLowerCase().indexOf(str.toString().toLowerCase()) >= 0)) {
      matches[matches.length] = $scope.people[i];
    }
  }
  return matches;
}
```

Let me know if it works or not.

**Comment by ghiden** [comment]
@hardiksondagar Did it work?

**Comment by ghiden** [comment]
Released v2.2.0.
